cms/djangoapps/contentstore/views/course.py

```diff
@@ -277,6 +281,22 @@ def course_handler(request, course_key_string=None):
         raise Http404
 
 
+@login_required
+def course_archive_handler(request, course_key_string=None):
+    if request.method == 'GET':
+        if course_key_string:
+            course_key = CourseKey.from_string(course_key_string)
+            course_details = CourseDetails.fetch(course_key)
+            end_date_setting = {
```

Comment:
We need to add `start_date` because otherwise this method clears the start date and next time a user tries to view the course, the page crashes as it tries to get the start date to show in the top menu. Also, `intro_video` needs to be added because this method directly updates this field in the course settings without checking if it exists in the passed dict hence leading to a crash.